Sources/Vertex.swift

```diff
@@ -47,23 +47,25 @@ public struct Vertex: Hashable {
 }
 
 extension Vertex: Codable {
-    private enum CodingKeys: CodingKey {
-        case position, normal, texcoord
+    private enum CodingKeys: String, CodingKey {
+        case position = "p"
+        case normal = "n"
+        case texcoord = "uv"
     }
 
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        let position = try container.decode(Vector.self, forKey: .position)
-        let normal = try container.decode(Vector.self, forKey: .normal)
-        let texcoord = try container.decodeIfPresent(Vector.self, forKey: .texcoord) ?? .zero
-        self.init(position, normal, texcoord)
+        position = try container.decode(Vector.self, forKey: .position)
+        normal = try container.decode(Vector.self, forKey: .normal).normalized()
+        texcoord = try container.decode(Vector.self, forKey: .texcoord)
+        
     }
 
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         try container.encode(position, forKey: .position)
         try container.encode(normal, forKey: .normal)
-        try texcoord == .zero ? () : container.encode(normal, forKey: .texcoord)
+        try container.encode(texcoord, forKey: .texcoord)
```

Comment:
Again, this seems to increase the size of untextured meshes for no obvious reason?